ipfsconn/ipfshttp/ipfshttp_test.go

```diff
@@ -99,24 +100,57 @@ func testPin(t *testing.T, method string) {
 	case "refs":
 		ipfs.config.PinTimeout = 1 * time.Second
 		c3 := test.SlowCid1
-		err = ipfs.Pin(ctx, c3, -1)
+		err = ipfs.Pin(ctx, api.PinCid(c3))
 		if err == nil {
 			t.Error("expected error pinning cid")
 		}
 	case "pin":
 		ipfs.config.PinTimeout = 5 * time.Second
 		c4 := test.SlowCid1
-		err = ipfs.Pin(ctx, c4, -1)
+		err = ipfs.Pin(ctx, api.PinCid(c4))
 		if err == nil {
 			t.Error("expected error pinning cid")
 		}
 	default:
 	}
 }
 
+func testPinUpdate(t *testing.T) {
+	ctx := context.Background()
+	ipfs, mock := testIPFSConnector(t)
+	defer mock.Close()
+	defer ipfs.Shutdown(ctx)
+
+	pin := api.PinCid(test.Cid1)
+	pin.PinUpdate = test.Cid1
+	// enforce pin/update even though it would be skipped
+	ipfs.config.PinMethod = "update"
```

Comment:
This should work without setting PinMethod

`ipfs.config.PinMethod` should remain the same for one daemon run. Plus, `refs` and `pin` are something that you can use for all pin operation, but `update` is useful only for a kind of pin operation. Meaning `update` is the value you need only for one pin operation.

Reply:
Please read the test. The connector will only attempt to use `pin/update` when the from-argument is pinned. This prevents situations where a pin/update failed (pin error) and it is recovered at a later moment, after the from-pin has been unpinned. That would result in an unrecoverable pin.

But the simplest way to check that the connector hits the "pin/update" endpoint (the purpose here), was to hit it in a way that fails in a expected manner. Therefore I need to generate an error by using pin/update when the `from` Cid is not pinned yet and the way to do it is this. After that the test runs the normal workflow by resetting.

Yes, it's a hack. The alternative is to launch a custom ipfs mock that I can inspect (making sure the calls hit the right endpoint). I can do that too if you think it's better... (it's more code and an extra mock but makes the test less hacky).